shared/database/source_control/working_dir.py

```diff
@@ -278,13 +279,27 @@ def verify_directory_in_project(session, project, directory_id):
         Other ways we could do this but this seems to work as a basic check
         Future perhaps permissions on individual directories
         """
-        directory = session.query(WorkingDir).filter(
-            WorkingDir.project_id == project.id,
-            WorkingDir.id == directory_id).first()
-        if directory:
+        
+        directory = WorkingDir.get(session, directory_id, project.id)
+        return bool(directory)
+    
+    # TODO: add tests
+    def can_member_view_datasets(session, project, member, dataset_ids):
+        # TODO: How come we do this inline?
+        from shared.database.permissions.roles import ValidObjectTypes
```

Comment:
Curious why this is done inline vs top-level (copied from other Policy checks in this class)